registry/registry.go

```diff
@@ -3,22 +3,25 @@ package registry
 import (
 	"go-bootcamp/infrastructure/database"
 	"go-bootcamp/interface/controller"
+	"go-bootcamp/usecase/clients"
 )
 
 type registry struct {
-	db database.DB
+	db            database.DB
+	pokeAPIConfig clients.PokeAPIConfig
 }
 
 type Registry interface {
 	NewAppController() controller.AppController
 }
 
-func New(db database.DB) Registry {
-	return &registry{db}
+func New(db database.DB, pac clients.PokeAPIConfig) Registry {
+	return &registry{db, pac}
 }
 
 func (r *registry) NewAppController() controller.AppController {
 	return controller.AppController{
 		Pokemon: r.NewPokemonController(),
+		PokeAPI: r.NewPokeAPIController(),
```

Comment:
The ideal scenario is not to keep creating Controllers as this two should be the same, What I mean is pokemon Api you are consuming should be used inside your already created controllers? why keep creating controllers, be careful as challenge specified this may require a refactoring, this means that your API from delivery 1 may or may not be the same.